controller/src/test/java/io/pravega/controller/store/stream/StreamTest.java

```diff
@@ -0,0 +1,129 @@
+        testStream(stream);
+    }
+
+    private void testStream(PersistentStreamBase<Integer> stream) throws InterruptedException, ExecutionException {
+        long creationTime1 = System.currentTimeMillis();
+        long creationTime2 = creationTime1 + 1;
+        final ScalingPolicy policy1 = ScalingPolicy.fixed(5);
+        final ScalingPolicy policy2 = ScalingPolicy.fixed(6);
+
+        final StreamConfiguration streamConfig1 = StreamConfiguration.builder().scope("test").streamName("test").scalingPolicy(policy1).build();
+        final StreamConfiguration streamConfig2 = StreamConfiguration.builder().scope("test").streamName("test").scalingPolicy(policy2).build();
+
+        CreateStreamResponse response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        stream.storeCreationTimeIfNotExists(creationTime1).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+
+        stream.createConfigurationIfNotExists(streamConfig1).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.EXISTS_CREATING, response.getStatus());
+
+        stream.createStateIfNotExists(State.UNKNOWN).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime2).get();
+        assertEquals(CreateStreamResponse.CreateStatus.EXISTS_CREATING, response.getStatus());
+
+        stream.updateState(State.CREATING).get();
+
+        response = stream.checkStreamExists(streamConfig1, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
+        response = stream.checkStreamExists(streamConfig2, creationTime1).get();
+        assertEquals(CreateStreamResponse.CreateStatus.NEW, response.getStatus());
```

Comment:
Since we have not stored the configuration in the previous run, this is treated as new stream and hence we are returning "New".
Next time we have saved the configuration. So since a configuration exists (we dont compare configurations, the mere fact that an older configuration exists means we will use it) we return EXISTS_CREATING